go/vt/vtgate/evalengine/compiler_test.go

```diff
@@ -715,6 +726,12 @@ func TestCompilerSingle(t *testing.T) {
 				t.Fatalf("bad collation evaluation from eval engine: got %d, want %d", expected.Collation(), tc.collation)
 			}
 
+			if tc.typeWanted.Type() != sqltypes.Unknown {
+				typ, err := env.TypeOf(converted)
+				require.NoError(t, err)
+				require.EqualValues(t, tc.typeWanted, typ)
```

Comment:
Since https://github.com/vitessio/vitess/pull/15783 changed `ctype` as well, not sure if this comparison is exactly correct now with how it would compare. You might want to use `ctype#Equal` now that is available to compare types.